### PR TITLE
New collection endpoints

### PR DESCRIPTION
**Pull request by Reettaphant**
New endpoint now prunes unnecessary metadata on save. We don't need to worry about the weird live/draft format behaviour if we don't allow edits to live collections - the publish endpoint will take care of all of this.

### CODE REVIEW

File: app/controllers/FaciaToolV2Controller.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    val v2Update: Option[V2Update] = request.body.asJson.flatMap(jsValue => jsValue.asOpt[V2Update])
+    v2Update match {
+      case Some(update) => {
+          val identity = request.user
+
+          updateActions.v2UpdateCollection(update.id, update.collection, identity)
+
+          faciaPress.press(PressCommand(
+            Set(update.id),
+            live = false,
```
Comment (RichieAHB) [inline app/controllers/FaciaToolV2Controller.scala]: Is this the bit where we're ignoring the weird live front editing from v1?

File: app/model/frontsapi.scala [hunk 2]
```diff
@@ -126,11 +130,27 @@ trait UpdateActionsTrait {
...
+    }
+
   def putMasterConfig(config: ConfigJson, identity: User): Option[ConfigJson] = {
     faciaApiIO.archiveMasterConfig(config, identity)
     faciaApiIO.putMasterConfig(config)
   }
 
+  def v2UpdateCollection(id: String, collection: CollectionJson, identity: User): CollectionJson = {
+    val collectionWithGroupsRemoved = removeGroupIfNoLongerGrouped(id, collection)
+    val prunedCollection = pruneBlock(collectionWithGroupsRemoved)
```
Comment (RichieAHB) [inline app/model/frontsapi.scala]: What does this line actually prune in reality? What sort of things are removed (or more to the point, what might we put here on the client that isn't wanted)?

File: app/controllers/FaciaToolV2Controller.scala (hunk 1 above)
Comment (Reettaphant) [reply app/controllers/FaciaToolV2Controller.scala]: Yes

File: app/model/frontsapi.scala (hunk 2 above)
Comment (Reettaphant) [reply app/model/frontsapi.scala]: This removes empty metadata objects and removes empty  group objects if the collection is no longer grouped. The only thing I could think of why a collection is not grouped anymore is if someone changes the layout. These transformations are not really changing anything in the collection, just cleaning up empty stuff.

Comment (RichieAHB) [reply app/model/frontsapi.scala]: 👍 makes perfect sense

**Comment by Reettaphant** [comment]
Commits were cherry-picked to another branch

**Pull request closed without merging**